static/js/util/validation.js

```diff
@@ -252,12 +257,10 @@ export function validateMonth(string) {
 
 /**
  * Validate a year string is an integer and fits into YYYY
- * @param {String} string The input string
- * @returns {Number|undefined} The valid year if a valid year value or undefined if not valid
  */
-export function validateYear(string) {
+export function validateYear(string: string): ?number {
   let year = filterPositiveInt(string);
-  if (year === undefined) {
+  if (year == null) { // eslint-disable-line eqeqeq
```

Comment:
It was necessary because of the return type of `filterPositiveInt`, which was `?number`. the `?number` type is a shorthand for `number|null|void`, so the typechecker was complaining about how we weren't checking for the case when `year` is `null` in addition to the case in which is undefined.

I just checked, and if I change the return type of `filterPositiveInt` from `?number` to `number|null` I think we can keep this as it was.